cicd/3-app/javabuilder/template.yml.erb

```diff
@@ -893,6 +899,107 @@ Resources:
               Expression: ANOMALY_DETECTION_BAND(m1, 8)
         ThresholdMetricId: ad1
 
+  <%=name%>MinimumUsageAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+        AlarmName: !Sub "${SubDomainName}_<%=name.downcase%>_minimum_usage"
+        AlarmDescription: This alarm is to be used as part of a composite alarm, not by itself.
+            It triggers if the usage is above a minimum threshold, so we do not alarm on error
+            rates if we have very low usage.
+        ActionsEnabled: false
+        MetricName: Invocations
+        Namespace: AWS/Lambda
+        Statistic: Sum
+        Dimensions:
+            - Name: FunctionName
+              Value: !Ref BuildAndRunJava<%=name%>ProjectFunction
+        Period: 300
+        EvaluationPeriods: 4
+        DatapointsToAlarm: 4
+        Threshold: 100
+        ComparisonOperator: GreaterThanOrEqualToThreshold
+        TreatMissingData: notBreaching
+  
+  <%=name%>ElevatedSevereErrorRateAlarm:
```

Comment:
To clarify - maybe I'm misunderstanding, but should these names be flipped? Looks like this alarms on the 10% threshold vs the one below alarms on the 90%. I figured "Elevated" would refer to the 90% alarm?

Reply:
Naming struggles here. I wanted to use "High" for the 90% alarms, but I couldn't re-use the name we had before because it doesn't let you change the type of an object in cloudformation (in this case, changing from alarm to composite alarm). Your suggestion makes sense, if we go with "Elevated" for the 90% alarm does it make sense for the 10/25% threshold to just be "error rate" or is there another adjective we could use?

Reply:
I swapped the names around for now, I think that makes more sense.